packages/sx05re/emuelec/gptokeyb/gptokeyb.cpp

```diff
@@ -222,255 +519,20 @@ int main(int argc, char* argv[])
-              emitKey(KEY_LEFT, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_DPAD_UP:
-              emitKey(KEY_UP, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_DPAD_RIGHT:
-              emitKey(KEY_RIGHT, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_DPAD_DOWN:
-              emitKey(KEY_DOWN, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_A:
-              emitKey(KEY_ENTER, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_B:
-              emitKey(KEY_ESC, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_BACK: // aka select
-              emitKey(KEY_PLAYPAUSE, is_pressed);
-              break;
-
-            case SDL_CONTROLLER_BUTTON_START:
-              emitKey(KEY_ENTER, is_pressed);
-              break;
-          }
-        } //kill mode
-      } break;
-
-      case SDL_CONTROLLERAXISMOTION:
-        if (xbox360_mode) {
-          switch (event.caxis.axis) {
-            case SDL_CONTROLLER_AXIS_LEFTX:
-              emitAxisMotion(ABS_X, event.caxis.value);
-              break;
-
-            case SDL_CONTROLLER_AXIS_LEFTY:
-              emitAxisMotion(ABS_Y, event.caxis.value);
-              break;
-
-            case SDL_CONTROLLER_AXIS_RIGHTX:
-              emitAxisMotion(ABS_RX, event.caxis.value);
-              break;
-
-            case SDL_CONTROLLER_AXIS_RIGHTY:
-              emitAxisMotion(ABS_RY, event.caxis.value);
-              break;
-
-            case SDL_CONTROLLER_AXIS_TRIGGERLEFT:
-              // The target range for the triggers is 0..255 instead of
-              // 0..32767, so we shift down by 7 as that does exactly the
-              // scaling we need (32767 >> 7 is 255)
-              emitAxisMotion(ABS_Z, event.caxis.value >> 7);
-              break;
-
-            case SDL_CONTROLLER_AXIS_TRIGGERRIGHT:
-              emitAxisMotion(ABS_RZ, event.caxis.value >> 7);
-              break;
-          }
-        } // xbox mode
-        break;
-      case SDL_CONTROLLERDEVICEADDED:
-        if (xbox360_mode == true || openbor_mode == true) {
-          SDL_GameControllerOpen(0);
-          /* SDL_GameController* controller = SDL_GameControllerOpen(0);
-       if (controller) {
-                        const char *name = SDL_GameControllerNameForIndex(0);
-                            printf("Joystick %i has game controller name '%s'\n", 0, name);
-                    }
-    */
-        } else {
-          SDL_GameControllerOpen(event.cdevice.which);
-        }
-        break;
-
-      case SDL_CONTROLLERDEVICEREMOVED:
-        if (
-          SDL_GameController* controller =
-            SDL_GameControllerFromInstanceID(event.cdevice.which)) {
-          SDL_GameControllerClose(controller);
-        }
-        break;
+      emitMouseMotion(mouseX, mouseY);
+      SDL_Delay(100);
```

Comment:
Maybe instead of hardcoding `100` here, we should have a constant `FAKE_MOUSE_RATE` for easier tweaking

Reply:
I tested this with `16` instead and the mouse movement is much better, CPU usage is not affected by the change (using htop both use 0.7% when the mouse emulation is used)